cypress/integration/pages/onDemandRadio/testsForCanonicalOnly.js

```diff
@@ -12,5 +14,45 @@ export default ({ service, pageType }) => {
         });
       }
     });
+    describe('Radio Schedule', () => {
+      it('should be displayed if there is enough schedule data', () => {
+        const isRadioScheduleOnPage = serviceHasRadioSchedule({
+          service,
+          variant,
+        });
+        cy.log(
+          `On Demand Radio Page configured for Radio Schedule? ${isRadioScheduleOnPage}`,
+        );
+
+        if (isRadioScheduleOnPage) {
+          const schedulePath = Cypress.env('currentPath');
+          const index = schedulePath.lastIndexOf('_radio/');
+          let newUrl = `${schedulePath.substring(
```

Comment:
The syntax and variable names here are a bit mad. It will also always append `?renderer_env=live` regardless of environment.

My suggestion would be something along these lines (comments below code show state of url being constructed). The way of getting `pathWithoutPid` still feels nasty - no doubt folk with more knowledge will have a nicer way of doing it.

```js
          const currentPath = Cypress.env('currentPath');
          // '/afaanoromoo/bbc_afaanoromoo_radio/programmes/w13xttnw?renderer_env=live'
          
          const pathWithoutPid = `${currentPath.split('_radio')[0]}_radio`;
          // '/afaanoromoo/bbc_afaanoromoo_radio'

          let schedulePath = `${pathWithoutPid}/schedule.json`.replace(
            'bbc_afaanoromoo_radio',
            'bbc_oromo_radio',
          );
          // '/afaanoromoo/bbc_oromo_radio/schedule.json'

          if (Cypress.env('APP_ENV') === 'test') {
            schedulePath += '?renderer_env=live';
          }
          // '/afaanoromoo/bbc_afaanoromoo_radio/schedule.json?renderer_env=live'

          cy.request(schedulePath).then(({ body: scheduleJson }) => {
          ...
```

Reply:
```
const currentPath = Cypress.env('currentPath');

const masterBrand = currentPath.split('/')[2]

let schedulePath = `${service}/${masterBrand}/schedule.json`.replace(
            'bbc_afaanoromoo_radio',
            'bbc_oromo_radio',
          );

```

Looks like you have the service value from the test description